pkg/checkpoint/secret.go

```diff
@@ -50,14 +59,84 @@ func (c *checkpointer) checkpointSecret(namespace, podName, secretName string, u
 
 	// TODO(aaron): No need to store if already exists
 	for f, d := range secret.Data {
-		if err := writeAndAtomicRename(filepath.Join(basePath, f), d, uid, gid, 0600); err != nil {
+		if err := writeAndAtomicRename(filepath.Join(basePath, f), d, uid, gid, defaultMode); err != nil {
 			return "", fmt.Errorf("failed to write secret %s: %v", secret.Name, err)
 		}
 	}
 
 	return basePath, nil
 }
 
+// checkpointSecretProjection will locally store secret data from volumes with ProjectedVolumeSources.
+// The path to the secret data becomes: checkpointSecretPath/namespace/podname/volumename/secret.file
```

Comment:
The logic below is a little hard to follow, but is this right?

```
checkpointSecretPath/namespace/podname/volumename/secret.file
```

I would have expected a constructed folder here, not a secret file.

```
checkpointSecretPath/namespace/podname/volumename/( directory with projected secrets and configmaps )
```

Reply:
the projected volume specifies the path that each key is written to which can include folder names. this is mentioned on the next line

https://kubernetes.io/docs/concepts/storage/volumes/#projected